File: hash-table/group-anagrams.cpp
class Solution {
public:
   
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
       // 1. 定义那个“抽屉柜”
        // 标签是 string，抽屉里装的是一堆 string
        unordered_map<string, vector<string>> mp;
        for(string s: strs){
            string label=s;
            sort(label.begin(),label.end());
            mp[label].push_back(s);
        }
        vector<vector<string>> result;
        for (auto it = mp.begin(); it != mp.end(); it++) {
            // it->second 就是抽屉里的那个 vector（那一组单词）
            result.push_back(it->second);
        }
        return result;
        
    }
};

File: sorting/group-anagrams.py
class Solution(object):
    def groupAnagrams(self, strs):
        """
        :type strs: List[str]
        :rtype: List[List[str]]
        """
        # 字母异位词是通过重新排列不同单词或短语的字母而形成的单词或短语，并使用所有原字母一次。
        seen={}
        for s in strs:
            key=''.join(sorted(s))
            if key not in seen:
                seen[key]=[]
            seen[key].append(s)
        return list(seen.values())
        